ref-arch/real-time-analytics.rst

```diff
@@ -0,0 +1,509 @@
+.. _introduction:
+
+Real Time Analytics
+#####################
+
+Over the last few years we've helped many different kinds of clients use Citus and noticed
+a technical problem many businesses have: running real-time analytics over large streams
+of data.
+
+For example, say you're building an HTTP analytics dashboard. You have enough clients that
+around every millisecond a user hits one of their websites and sends a log record to you.
+You want to ingest all of those records (1000 inserts/sec) and create a dashboard which
+shows your clients things like how many requests to their sites are errors. It's important
+that this data show up with as little latency as possible so your clients can fix problems
+with their sites. It's also useful to show graphs of historical data, however keeping all
+the raw data around forever is prohibitively expensive.
+
+Or maybe you're building an advertising network and want to show clients clickthrough
+rates on their campaigns. In this example latency is also critical, raw data volume is
+also high, and both historical and live data are important.
+
+In this reference architecture we'll demonstrate how to build part of the first example
+but this architecture would work equally well for the second and many other business
+use-cases.
+
+Running It Yourself
+-------------------
+
+There's `a github repo <https://github.com>`_ with scripts and usage instructions. If
+you've gone through our installation instructions for running on either single or multiple
+machines you're ready to try it out. There will be code snippets in this tutorial but they
+don't quite specify a complete system, the github repo has all the details in one place.
+
+Data Model
+----------
+
+The data we're dealing with is an immutable stream of log data. Here we'll insert directly
+into Citus but it's also common for this data to first be routed through something like
+Kafka. Doing so makes the system a little more resilient to failures, lets the data be
+routed to multiple places (such as a warehouse like redshift), and (once data volumes
```

Comment:
Maybe these side notes in parentheses are not so important and could be omitted, or rewritten to something like:

makes it a little easier to start pre-aggregating the data before inserting in case raw data volumes become too high to query at run-time.